### PR TITLE
@maker/esm update

### PR DESCRIPTION
**Pull request by CeliktepeMurat**
This PR contains updates that requested by Forta team for ESM modules contract

### CODE REVIEW

File: Maker-Agents-Suit-Module/ESMAgent/src/ESM.join.event.agent.ts [hunk 1]
```diff
@@ -36,8 +48,25 @@ const provideESMJoinEventAgent = (
...
+
+    if (logs.length) {
+      findingGenerator = createFindingGenerator(
+        _alertID,
+        logs[0].topics[1],
+        BigInt(logs[0].data).toString(),
+      );
+    } else {
+      findingGenerator = createFindingGenerator(_alertID, '', '');
+    }
```
Comment (e1Ru1o) [inline Maker-Agents-Suit-Module/ESMAgent/src/ESM.join.event.agent.ts:66]: This works only if the transaction has only one Join event log. I am going to improve the `provideEventCheckerHandler` to pass the data to the finding generator.

File: Maker-Agents-Suit-Module/ESMAgent/src/ESM.join.event.agent.ts [hunk 2]
```diff
@@ -13,21 +14,28 @@ import {
...
-      alertId: alertID,
+      alertId: _alertID,
+      protocol: 'Maker',
       severity: FindingSeverity.Medium,
-      type: FindingType.Unknown,
+      type: FindingType.Suspicious,
+      everestId: MAKER_EVEREST_ID,
+      metadata: {
+        usr: metadata!.topics[1],
+        amount: BigInt(metadata!.data).toString(),
```
Comment (e1Ru1o) [inline Maker-Agents-Suit-Module/ESMAgent/src/ESM.join.event.agent.ts:37]: You should decode this values before printing it

**Review by e1Ru1o — CHANGES_REQUESTED** [review]

File: Maker-Agents-Suit-Module/ESMAgent/src/ESM.join.event.agent.ts [hunk 3]
```diff
@@ -10,24 +11,31 @@ import {
...
-      alertId: alertID,
+      alertId: _alertID,
+      protocol: 'Maker',
       severity: FindingSeverity.Medium,
-      type: FindingType.Unknown,
+      type: FindingType.Suspicious,
+      everestId: MAKER_EVEREST_ID,
+      metadata: {
+        usr: decodeParam('address', metadata!.topics[1]).toLowerCase(),
+        amount: decodeParam('uint256', metadata!.data).toLowerCase(),
```
Comment (e1Ru1o) [inline Maker-Agents-Suit-Module/ESMAgent/src/ESM.join.event.agent.ts:37]: You can remove the `.toLowerCase()` in `amount`.
```suggestion
        amount: decodeParam('uint256', metadata!.data),
```

**Review by e1Ru1o — COMMENTED** [review]
I think that you should change your tests. 

Your tests currently looks like:
```javascript
const tx1 = new TestTransactionEvent().addEventLog(...params1);
const tx2 = new TestTransactionEvent().addEventLog(...params2);

const findings = [
  ...(await handle(tx1)),
  ...(await handle(tx2)),
];
```
You should consider changing to:
```javascript
const tx = new TestTransactionEvent()
   .addEventLog(...params1)
   .addEventLog(...params2);

const findings = await handle(tx);
```
This is needed because you need to test that the handler is capable of reporting the two types of events while handling a single transaction. All test can change to use a single transaction.

**Review by e1Ru1o — APPROVED** [review]

File: Maker-Agents-Suit-Module/ESMAgent/src/ESM.join.event.agent.ts [hunk 4]
```diff
@@ -10,24 +11,31 @@ import {
...
 export const MAKER_ESM_JOIN_EVENT_SIGNATURE = 'Join(address,uint256)';
 export const MKR_DECIMALS = 18;
+export const MAKER_EVEREST_ID = '0xbabb5eed78212ab2db6705e6dfd53e7e5eaca437';
 
 const filterLog = (log: Log): boolean => {
-  const value = BigInt(log.data) / BigInt(10 ** MKR_DECIMALS);
-
-  return value > 2;
+  const amount = decodeParam('uint256', log.data);
+  return amount > 2;
```
Comment (mauricio1802) [inline Maker-Agents-Suit-Module/ESMAgent/src/ESM.join.event.agent.ts:22]: Can you use only `2` here? Don't you need to use the `MKR` decimals?

**Review by mauricio1802 — CHANGES_REQUESTED** [review]

**Review by mauricio1802 — APPROVED** [review]

**Pull request merged**